src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -5,190 +5,119 @@
 
+        def __getitem__(self, item):
+            return self.__d[item]
 
-def fixupSecondaryFileNames(process):
-    """
-    _fixupSecondaryFileNames_
+        def __call__(self, *args, **kwargs):
+            pass
 
-    Make sure that the process has a secondaryFileNames parameter.
+        def __repr__(self):
+            return "{module}.{name}".format(module=self.__module, name=self.__name)
 
-    """
-    if not hasattr(process.source, "secondaryFileNames"):
-        process.source.secondaryFileNames = cms.untracked.vstring()
-    return
+    return DummyClass
 
 
-def fixupFirstLumi(process):
+class Unpickler(pickle.Unpickler):
     """
-    _fixupFirstLumi
+    _Unpickler_
+    Use this when loading a PSet pickle
+    and a specific module is not available (i.e.: FWCore modules)
 
-    Make sure that the process has firstLuminosityBlock parameter.
     """
-    if not hasattr(process.source, "firstLuminosityBlock"):
-        process.source.firstLuminosityBlock = cms.untracked.uint32(1)
-    return
+    def find_class(self, module, name):
+        return factory(module, name)
 
 
 class SetupCMSSWPset(ScriptInterface):
     """
     _SetupCMSSWPset_
 
     """
-    fixupDict = {"process.GlobalTag.globaltag": fixupGlobalTag,
-                 "process.GlobalTag.DBParameters.transactionId": fixupGlobalTagTransaction,
-                 "process.source.fileNames": fixupFileNames,
-                 "process.source.secondaryFileNames": fixupSecondaryFileNames,
-                 "process.maxEvents.input": fixupMaxEvents,
-                 "process.source.skipEvents": fixupSkipEvents,
-                 "process.source.firstEvent": fixupFirstEvent,
-                 "process.source.firstRun": fixupFirstRun,
-                 "process.source.lastRun": fixupLastRun,
-                 "process.source.lumisToProcess": fixupLumisToProcess,
-                 "process.source.firstLuminosityBlock": fixupFirstLumi}
 
     def __init__(self, crabPSet=False):
         ScriptInterface.__init__(self)
         self.crabPSet = crabPSet
         self.process = None
         self.jobBag = None
         self.logger = logging.getLogger()
+        self.tweak = PSetTweak()
+        self.scram = None
+        self.configPickle = "Pset.pkl"
+        self.psetFile = None
+
+    def createScramEnv(self):
+        """
+        _createScramEnv_
+
+        Create scram environment
+
+        """
+        scram = Scram(
+            version=self.getCmsswVersion(),
+            directory=self.stepSpace.location,
+            architecture=self.getScramVersion(),
+            initialise=self.step.data.application.setup.softwareEnvironment
+        )
+        scram.project() # creates project area
+        scram.runtime() # creates runtime environment
+
+        return scram
+
+    def scramRun(self, cmdArgs):
+        """
+        _scramRun_
+
+        Run command inside scram environment
+
+        """
+        self.logger.info("ScramRun command args: %s", cmdArgs)
```

Comment:
Should we change this log to something like `Running CMSSW WM tools with args:`? Unless this method can execute things different than cmssw-wm-tools scripts(?)
Similar change would apply in case it returns with a code != 0

Reply:
Technically, it can run any command inside the scram environment. Currently,  it  looks like this:

```
  2021-04-09 13:52:18,916:INFO:SetupCMSSWPset:ScramRun command args: edm_pset_pickler.py --input /tmpscratch/users/khurtado/work/validation/relVals/job/WMTaskSpace/cmsRun1/PSet.py --output_pkl /tmpscratch/users/khurtado/work/validation/relVals/job/WMTaskSpace/cmsRun1/PSet.pkl
```

Reply:
We leave it as it is then. Thanks